components/renku_data_services/users/db.py

```diff
@@ -65,16 +54,26 @@ async def get_user(self, requested_by: APIUser, id: str) -> UserInfo | None:
             stmt = select(UserORM).where(UserORM.keycloak_id == id)
             res = await session.execute(stmt)
             orm = res.scalar_one_or_none()
+            if not orm and id == requested_by.id:
+                return await self._add_api_user(requested_by)
```

Comment:
get_user inserting a user in the db class is a bit odd, shouldn't this be handled by the client or calling code? Seems like the kind of side-effect that could bite us in the future

Reply:
at the very least it should be called get_or_create_user or similar

Reply:
The reason for this is to avoid cases where a freshly signed up user hits the /user endpoint in the API and we return a 404 because the data services API has not synced with Keycloak yet. And the user will be considered to not exist for the next 2-3 minutes until the cronjob that syncs keyclaok and the data services fires and completes.

It is not that nice. But I have no idea how to avoid this problem for newly created users. If we do not have this in place we will definitely make it so that newly created users dont exist for a little while.

Reply:
I will rename the function though.